diff --git a/test.c b/test.c
index 5c5d94d..0a5eb8c 100644
--- a/test.c
+++ b/test.c
@@ -9,9 +9,15 @@ void bad() {
     arr[0] = 3;
     printf("%d %d %d", arr[0], arr[1], arr[2]);
     int val = 0;
-    val = arr[x+y+5];
+    if (x + y + 5 < n) {
+        val = arr[x + y + 5];
+    } else {
+        printf("Index out of bounds\n");
+    }
 }
 
 int main() {
     bad();
+    return 0;
+}
 }
\ No newline at end of file
